File: R/figures.R

plot_airport_line <- function(df, selected_airport){
  trafic_aeroports <- df %>%
    mutate(trafic = apt_pax_dep + apt_pax_tr + apt_pax_arr) %>%
    filter(apt %in% selected_airport) %>%
    mutate(
      date = as.Date(paste(an, mois, "01", sep="-"))
    )
  
  figure_plotly <- trafic_aeroports %>%
    plot_ly(
      x = ~date, y = ~trafic,
      text = ~apt_nom,
      hovertemplate = paste("<i>Aéroport:</i> %{text}<br>Trafic: %{y}") ,
      type = 'scatter', mode = 'lines+markers')
  
  return(figure_plotly)
}

File: src/figures.py
import plotly.express as px

def plot_airport_line(df, selected_airport):

  trafic_aeroports = (
    df
    .loc[df['apt'] == default_airport]
  )
  trafic_aeroports['date'] = pd.to_datetime(
    trafic_aeroports['anmois'] + '01', format='%Y%m%d'
  )


  fig = px.line(
    trafic_aeroports, x="date", y="trafic",
    text="apt_nom"
  )
  fig.update_traces(
    mode="markers+lines", type = "scatter",
    hovertemplate="<i>Aéroport:</i> %{text}<br>Trafic: %{y}"
  )

  return fig